Vaadin 15+ applications cannot be started from IntelliJ using the Application class

### Description of the bug
A Vaadin 14 application can be started by running the `Application` class, without running any Maven targets
A Vaadin 15+ application fails to start

### Minimal reproducible example
1. Open https://start.vaadin.com?preset=latest
2. Download
3. Import and run in IntelliJ

### Expected behavior
The app starts, like a Vaadin 14 app.

### Actual behavior
```
2020-08-24 13:59:36.224  INFO 66024 --- [  restartedMain] com.example.application.Application      : Starting Application on MacBook-Prime.local with PID 66024 (/Users/artur/Downloads/stable16/target/classes started by artur in /Users/artur/Downloads/stable16)
2020-08-24 13:59:36.229  INFO 66024 --- [  restartedMain] com.example.application.Application      : No active profile set, falling back to default profiles: default
2020-08-24 13:59:36.313  INFO 66024 --- [  restartedMain] .e.DevToolsPropertyDefaultsPostProcessor : Devtools property defaults active! Set 'spring.devtools.add-properties' to 'false' to disable
2020-08-24 13:59:36.314  INFO 66024 --- [  restartedMain] .e.DevToolsPropertyDefaultsPostProcessor : For additional web related logging consider setting the 'logging.level.web' property to 'DEBUG'
2020-08-24 13:59:37.364  INFO 66024 --- [  restartedMain] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8080 (http)
2020-08-24 13:59:37.375  INFO 66024 --- [  restartedMain] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2020-08-24 13:59:37.375  INFO 66024 --- [  restartedMain] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.27]
2020-08-24 13:59:37.475  INFO 66024 --- [  restartedMain] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2020-08-24 13:59:37.475  INFO 66024 --- [  restartedMain] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 1161 ms
2020-08-24 13:59:37.552 ERROR 66024 --- [  restartedMain] o.a.c.c.C.[Tomcat].[localhost].[/]       : Exception sending context initialized event to listener instance of class [com.vaadin.flow.spring.VaadinServletContextInitializer$VaadinAppShellContextListener]

java.lang.RuntimeException: Unable to initialize com.vaadin.flow.spring.VaadinServletContextInitializer$VaadinAppShellContextListener
	at com.vaadin.flow.spring.VaadinServletContextInitializer$FailFastServletContextListener.contextInitialized(VaadinServletContextInitializer.java:166) ~[vaadin-spring-13.0.0.jar:na]
	at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:4683) ~[tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5146) ~[tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183) ~[tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1384) ~[tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1374) ~[tomcat-embed-core-9.0.27.jar:9.0.27]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[na:na]
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75) ~[tomcat-embed-core-9.0.27.jar:9.0.27]
	at java.base/java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:140) ~[na:na]
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:909) ~[tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:841) ~[tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183) ~[tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1384) ~[tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1374) ~[tomcat-embed-core-9.0.27.jar:9.0.27]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[na:na]
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75) ~[tomcat-embed-core-9.0.27.jar:9.0.27]
	at java.base/java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:140) ~[na:na]
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:909) ~[tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:262) ~[tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183) ~[tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.core.StandardService.startInternal(StandardService.java:421) ~[tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183) ~[tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:930) ~[tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183) ~[tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.startup.Tomcat.start(Tomcat.java:459) ~[tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.springframework.boot.web.embedded.tomcat.TomcatWebServer.initialize(TomcatWebServer.java:107) ~[spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.web.embedded.tomcat.TomcatWebServer.<init>(TomcatWebServer.java:88) ~[spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.web.embedded.tomcat.TomcatServletWebServerFactory.getTomcatWebServer(TomcatServletWebServerFactory.java:438) ~[spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.web.embedded.tomcat.TomcatServletWebServerFactory.getWebServer(TomcatServletWebServerFactory.java:191) ~[spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.createWebServer(ServletWebServerApplicationContext.java:180) ~[spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.onRefresh(ServletWebServerApplicationContext.java:153) ~[spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:544) ~[spring-context-5.2.0.RELEASE.jar:5.2.0.RELEASE]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:141) ~[spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:747) ~[spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:397) ~[spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:315) ~[spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1226) ~[spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1215) ~[spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at com.example.application.Application.main(Application.java:14) ~[classes/:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:567) ~[na:na]
	at org.springframework.boot.devtools.restart.RestartLauncher.run(RestartLauncher.java:49) ~[spring-boot-devtools-2.2.0.RELEASE.jar:2.2.0.RELEASE]
Caused by: java.lang.IllegalStateException: There are neither 'flow-build-info.json' nor 'webpack.config.js' file available in the project/working directory. Ensure 'webpack.config.js' is present or trigger creation of 'flow-build-info.json' via running 'prepare-frontend' Maven goal.
	at com.vaadin.flow.server.DeploymentConfigurationFactory.verifyMode(DeploymentConfigurationFactory.java:416) ~[flow-server-3.0.0.jar:3.0.0]
	at com.vaadin.flow.server.DeploymentConfigurationFactory.readBuildInfo(DeploymentConfigurationFactory.java:276) ~[flow-server-3.0.0.jar:3.0.0]
	at com.vaadin.flow.server.DeploymentConfigurationFactory.createInitParameters(DeploymentConfigurationFactory.java:165) ~[flow-server-3.0.0.jar:3.0.0]
	at com.vaadin.flow.server.DeploymentConfigurationFactory.createPropertyDeploymentConfiguration(DeploymentConfigurationFactory.java:127) ~[flow-server-3.0.0.jar:3.0.0]
	at com.vaadin.flow.spring.VaadinServletContextInitializer$SpringStubServletConfig.createDeploymentConfiguration(VaadinServletContextInitializer.java:819) ~[vaadin-spring-13.0.0.jar:na]
	at com.vaadin.flow.spring.VaadinServletContextInitializer$SpringStubServletConfig.createDeploymentConfiguration(VaadinServletContextInitializer.java:796) ~[vaadin-spring-13.0.0.jar:na]
	at com.vaadin.flow.spring.VaadinServletContextInitializer$SpringStubServletConfig.access$700(VaadinServletContextInitializer.java:726) ~[vaadin-spring-13.0.0.jar:na]
	at com.vaadin.flow.spring.VaadinServletContextInitializer$VaadinAppShellContextListener.failFastContextInitialized(VaadinServletContextInitializer.java:423) ~[vaadin-spring-13.0.0.jar:na]
	at com.vaadin.flow.spring.VaadinServletContextInitializer$FailFastServletContextListener.contextInitialized(VaadinServletContextInitializer.java:162) ~[vaadin-spring-13.0.0.jar:na]
	... 43 common frames omitted

2020-08-24 13:59:37.553 ERROR 66024 --- [  restartedMain] o.apache.catalina.core.StandardContext   : One or more listeners failed to start. Full details will be found in the appropriate container log file
2020-08-24 13:59:37.553 ERROR 66024 --- [  restartedMain] o.apache.catalina.core.StandardContext   : Context [] startup failed due to previous errors
2020-08-24 13:59:37.590  INFO 66024 --- [  restartedMain] o.apache.catalina.core.StandardService   : Stopping service [Tomcat]
2020-08-24 13:59:37.596  WARN 66024 --- [  restartedMain] ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.context.ApplicationContextException: Unable to start web server; nested exception is org.springframework.boot.web.server.WebServerException: Unable to start embedded Tomcat
```

--- a/flow-server/src/main/java/com/vaadin/flow/server/DeploymentConfigurationFactory.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/DeploymentConfigurationFactory.java
@@ -181,13 +181,6 @@ public final class DeploymentConfigurationFactory implements Serializable {
                 initParameters.put(FALLBACK_CHUNK, fallbackChunk);
             }
         }
-
-        try {
-            verifyMode(json != null, hasWebpackConfig(initParameters));
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
-        }
-
     }
 
     private static void setInitParametersUsingTokenData(
@@ -429,42 +422,6 @@ public final class DeploymentConfigurationFactory implements Serializable {
             String message = String.format(DEV_FOLDER_MISSING_MESSAGE, folder);
             throw new IllegalStateException(message);
         }
-    }
-
-    private static void verifyMode(boolean hasTokenFile,
-            boolean hasWebpackConfig) {
-
-        if (!hasTokenFile && !hasWebpackConfig) {
-            // If no flow-build-info.json file exists, and no appropriate
-            // webpack.config.js is found in the current working directory, then
-            // show an error message that suggest either triggering creation of
-            // flow-build-info.json or ensuring webpack.config.js is present in
-            // the working directory.
-            throw new IllegalStateException(ERROR_DEV_MODE_NO_FILES);
-        }
-
-        // If flow-build-info.json doesn't exist, but an appropriate
-        // webpack.config.js is found in the working directory, then launch a
-        // dev server with configuration based on the project/working directory
-        // location
-        if (!hasTokenFile && hasWebpackConfig) {
-            // the current working directory will be used automatically by the
-            // dev server unless it's specified explicitly
-            logger.warn(
-                    "Found 'webpack.config.js' in the project/working directory. "
-                            + "Will use it for webpack dev server.");
-        }
-    }
-
-    private static boolean hasWebpackConfig(Properties initParameters)
-            throws IOException {
-        String baseDir = initParameters
-                .getProperty(FrontendUtils.PROJECT_BASEDIR);
-        File projectBaseDir = baseDir == null ? new File(".")
-                : new File(baseDir);
-        File webPackConfig = new File(projectBaseDir,
-                FrontendUtils.WEBPACK_CONFIG);
-        return FrontendUtils.isWebpackConfigFile(webPackConfig);
     }
 
     private static void readUiFromEnclosingClass(

